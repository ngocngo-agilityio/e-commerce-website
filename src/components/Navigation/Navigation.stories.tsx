import type { Meta, StoryObj } from '@storybook/react';
import { fn } from '@storybook/test';
import Navigation from '.';

const meta = {
  title: 'Navigation',
  component: Navigation,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    backgroundColor: { control: 'color' },
  },
  args: { onClick: fn() },
} satisfies Meta<typeof Navigation>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
