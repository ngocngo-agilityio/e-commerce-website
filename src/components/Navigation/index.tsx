// Libs
import { Flex, Text } from '@chakra-ui/react';
import Link from 'next/link';

// Constants
import { NAVIGATION } from '@constants';

const Navigation = (): JSX.Element => {
  return (
    <Flex as="nav" gap="30px" wrap="wrap">
      {NAVIGATION.map((item, index) => {
        const { title, path } = item;

        return (
          <Link key={`navigation-header-${index}`} href={path}>
            <Text
              as="span"
              textTransform="uppercase"
              fontFamily="arimo"
              fontWeight="bold"
              size="xs"
              color="navigation.color"
            >
              {title}
            </Text>
          </Link>
        );
      })}
    </Flex>
  );
};

export default Navigation;
