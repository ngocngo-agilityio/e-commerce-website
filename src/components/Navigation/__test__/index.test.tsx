// Libs
import { render } from '@testing-library/react';

// Components
import Navigation from '..';

describe('Navigation component', () => {
  test('should render Navigation successfully', () => {
    const { container } = render(<Navigation />);

    expect(container).toBeInTheDocument();
  });

  test('should match snapshot for Navigation', () => {
    const { container } = render(<Navigation />);

    expect(container).toMatchSnapshot();
  });
});
