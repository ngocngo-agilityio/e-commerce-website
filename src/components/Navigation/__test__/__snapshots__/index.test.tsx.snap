// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should match snapshot for Navigation 1`] = `
<div>
  <nav
    class="css-7dibxd"
  >
    <a
      href="/"
    >
      <span
        class="chakra-text css-1fd5thk"
      >
        Home
      </span>
    </a>
    <a
      href="#"
    >
      <span
        class="chakra-text css-1fd5thk"
      >
        About
      </span>
    </a>
    <a
      href="#"
    >
      <span
        class="chakra-text css-1fd5thk"
      >
        Contact us
      </span>
    </a>
  </nav>
</div>
`;
